tiledb/sm/filter/filter_buffer.cc

```diff
@@ -55,11 +55,7 @@ FilterBuffer::BufferOrView::BufferOrView(
       tdb_new(Buffer, (char*)buffer->data() + offset, nbytes));
 }
 
-FilterBuffer::BufferOrView::BufferOrView(BufferOrView&& other) {
-  underlying_buffer_.swap(other.underlying_buffer_);
-  view_.swap(other.view_);
-  std::swap(is_view_, other.is_view_);
```

Comment:
We were accessing `is_view_` while unititialized. A default move constructor suffices here.